generate_build_matrix.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
```

Comment:
Please add some comments/documentation to this file. I think it is not immediately obvious what the steps in this file do.